src/python/WMCore/Services/WorkQueue/WorkQueue.py

```diff
@@ -237,6 +237,31 @@ def cancelWorkflow(self, wf):
         elements = [x['id'] for x in data.get('rows', []) if x['key'][1] not in nonCancelableElements]
         return self.updateElements(*elements, Status='CancelRequested')
 
+    def updateSiteLists(self, wf, siteWhiteList, siteBlackList):
+        """
+        Update site lists of a workflow
+        :param wf: workflow name
+        :param siteWhiteList: new site white list
+        :param siteBlackList: new site black list
+        :return: None
+        """
+        # Update elements in Available status
+        data = self.db.loadView('WorkQueue', 'elementsDetailByWorkflowAndStatus',
+                                {'startkey': [wf], 'endkey': [wf, {}],
+                                 'reduce': False})
+        elementsToUpdate = [x['id'] for x in data.get('rows', [])]
```

Comment:
I am thinking that we should filter these elements by Available and perhaps Acquired status. For now, can you filter them by the former only? Further thought is needed.

Reply:
I don't know how to apply filter, I followed logic of `updatePriority` function. Please point me to an example how filters are applied within CouchDB.

Reply:
To filter it in the query, I think we would have to pass in a tuple with the (wflow_name, status). However, I actually think we should use a cheaper view like:
https://cmsweb.cern.ch/couchdb/workqueue/_design/WorkQueue/_view/jobStatusByRequest?reduce=false

then iterate over the rows and compare the `Status` field with what we want.